src/traces/heatmap/clean_2d_array.js

```diff
@@ -9,16 +9,17 @@
 'use strict';
 
 var isNumeric = require('fast-isnumeric');
+var Lib = require('../../lib');
 
-module.exports = function clean2dArray(zOld, transpose) {
+module.exports = function clean2dArray(zOld, trace, xa, ya) {
```

Comment:
I'm curious also, you have a bunch of `if(ax && ax[/* some key */])` statements in this routine. Is there a situation where `clearn2dArray` is called w/o `xa` and/or `ya`? 

If not, I'd just assume that they exist.

Reply:
> Is there a situation where `clearn2dArray` is called w/o `xa` and/or `ya`?

I think it did occur yes. Let me confirm that for you!